### PR TITLE
improve: rename `latestBlockNumber` to `latestBlockSearched`

### PR DESCRIPTION
**Pull request by nicholaspai**
The current name `latestBlockNumber` usually implies that it should return the HEAD block for the provider but the way its used, it 
should actually return the latest block searched by the client.

### CODE REVIEW

File: src/clients/AcrossConfigStoreClient/AcrossConfigStoreClient.ts [hunk 1]
```diff
@@ -76,7 +76,7 @@ export class AcrossConfigStoreClient extends BaseAbstractClient {
 
   protected rateModelDictionary: across.rateModel.RateModelDictionary;
   public firstBlockToSearch: number;
-  public latestBlockNumber = 0;
+  public latestBlockSearched = 0;
```
Comment (pxrl) [inline src/clients/AcrossConfigStoreClient/AcrossConfigStoreClient.ts:79]: Should we put this in `BaseAbstractClient.ts` ? This generic update functionality feels like it could belong there and would give greater consistency across all of the inheriting classes.

File: src/clients/HubPoolClient.ts [hunk 2]
```diff
@@ -107,7 +107,7 @@ export class HubPoolClient extends BaseAbstractClient {
     cachingMechanism?: CachingMechanismInterface
   ) {
     super(cachingMechanism);
-    this.latestBlockNumber = deploymentBlock === 0 ? deploymentBlock : deploymentBlock - 1;
+    this.latestBlockSearched = deploymentBlock === 0 ? deploymentBlock : deploymentBlock - 1;
```
Comment (pxrl) [inline src/clients/HubPoolClient.ts:110]: Opportunistic tweak:

```suggestion
    this.latestBlockSearched = Math.min(deploymentBlock - 1, 0);
```

File: src/utils/BlockUtils.ts
Comment (pxrl) [inline src/utils/BlockUtils.ts:1]: All of the changes in this file are unrelated to the {Hub,Spoke}PoolClient updates so they should be reverted.

File: src/clients/AcrossConfigStoreClient/AcrossConfigStoreClient.ts (hunk 1 above)
Comment (nicholaspai) [reply src/clients/AcrossConfigStoreClient/AcrossConfigStoreClient.ts]: you mean move these generic class properties like this? 7a07154d3f828583ef3c7a0dd577fa3cc474a48f

File: src/utils/BlockUtils.ts
Comment (nicholaspai) [reply src/utils/BlockUtils.ts:1]: I think these are correct though because this PR is essentially deprecating `latestBlockNumber` in favor of a variable named `latestBlockSearched`, thoughts?

Comment (pxrl) [reply src/utils/BlockUtils.ts:1]: I think something like `highBlock` or `endBlock` is a better fit, since block number in this context is only choosing the end block that's used for determining an average block time over a block range - it's not related to searching.

File: src/clients/AcrossConfigStoreClient/AcrossConfigStoreClient.ts (hunk 1 above)
Comment (pxrl) [reply src/clients/AcrossConfigStoreClient/AcrossConfigStoreClient.ts]: Yeah - the batched update feature is basically common between all of the classes, but I have noticed that we have some slight differences between them (and duplicated code, to some extent). We can improve on that by pulling it out to the common class that they all inherit from :+1:.

File: src/clients/BaseAbstractClient.ts [hunk 3]
```diff
@@ -6,6 +6,8 @@ import { isDefined } from "../utils";
  */
 export abstract class BaseAbstractClient {
   protected _isUpdated: boolean;
+  public firstBlockToSearch = 0;
+  public latestBlockSearched: number | undefined;
```
[the remark this answers is not in the document]
Comment (nicholaspai) [reply src/clients/BaseAbstractClient.ts:10]: Ok but we'll have to be careful to change any checks in relayer-v2 for this being undefined

[the remark this answers is not in the document]
Comment (nicholaspai) [reply src/clients/BaseAbstractClient.ts]: I'm not sure where I see that type assertion in `BundleUtils` but this commit makes `latestBlockSearched` type number d983bcb482bf2f0f62cdf7ae7473ab18a8623c2c

[the remark this answers is not in the document]
Comment (james-a-morris) [reply src/clients/BaseAbstractClient.ts]: This may be a good push to add the tsconfig option to check for undefined/null types in the relayer. We currently don't check for it

File: src/transfers-history/services/SpokePoolEventsQueryService.ts [hunk 4]
```diff
@@ -23,8 +23,8 @@ export class SpokePoolEventsQueryService {
     let speedUpDepositEvents: RequestedSpeedUpDepositEvent[] = [];
     let blockTimestampMap: { [blockNumber: number]: number } = {};
 
-    if (this.latestBlockNumber) {
-      from = this.latestBlockNumber + 1;
+    if (this.latestBlockSearched) {
```
Comment (james-a-morris) [inline src/transfers-history/services/SpokePoolEventsQueryService.ts:26]: Do we want to check if this is undefined, or is it okay to jump to the else statement if `latestBlockSearched === 0`?

My assumption is that we consider latestBlockSearched=0 an erroneous state, but it's worth double checking

**Review by james-a-morris — APPROVED** [review]
Nice refactor - I only have one high level question about the relayer accounting for null/undefined states.

File: src/utils/BlockUtils.ts
Comment (nicholaspai) [reply src/utils/BlockUtils.ts:1]: a5eb37cec01216a07db86193188d4095274dab55

thoughts

**Review by pxrl — APPROVED** [review]
LGTM!
